### PR TITLE
util/iptables: fix cross-build failures due to syscall.Flock()

### PR DESCRIPTION
**Pull request by dcbw**
Fixes: https://github.com/kubernetes/kubernetes/issues/45554
Related: https://github.com/kubernetes/kubernetes/pull/44895

@luxas @k82cn @danwinship

### CODE REVIEW

**Comment by vishh** [comment]
@dcbw can you run `hack/udpate-bazel.sh` again?

**Comment by vishh** [comment]
@dcbw this is blocking other PRs, can we get this through today?

**Comment by dcbw** [comment]
@vishh yeah, working on this...

hack/update-bazel.sh doesn't work because it doesn't correctly handle GOPATH like all the rest of the hack/ stuff does (eg, as a subpath of build/).  IIRC I tried to make it do that a couple months ago, but we can't because it does really odd things with looking at package imports.  So anyway, I'm stuck making the Jenkins stuff do the bazel verification.

**Comment by dcbw** [comment]
Test Failures
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/controllers/autoregister TestCheckAPIService 12s

seems to be a flake?

**Comment by dcbw** [comment]
@k8s-bot unit test this issue https://github.com/kubernetes/kubernetes/issues/43781

**Comment by luxas** [comment]
@deads2k known flake? ^

**Comment by luxas** [comment]
@dcbw I always update bazel like this:
```
docker run -it -v $(pwd):/go/src/k8s.io/kubernetes -w /go/src/k8s.io/kubernetes golang:1.7 /bin/bash -c "hack/update-bazel.sh"
```

have worked well for me for a long time

**Comment by dcbw** [comment]
@luxas thanks, that works!  I'll use that from now on.

**Comment by dcbw** [comment]
@luxas @vishh either of you know how to trigger a cross-build so we can make sure this fixes the issue?

**Comment by dcbw** [comment]
@k8s-bot pull-kubernetes-cross test this

**Comment by k82cn** [comment]
@dcbw , `golang.org/x/sys/unix` did not support in windows.

**Comment by dcbw** [comment]
@k82cn yeah, I thought that by the "use golang.org/x/syscall" the suggestion was to just use that.  I'd originally done a split out with build +linux, and I'll just do both things.

**Comment by dcbw** [comment]
@k8s-bot pull-kubernetes-cross test this

**Comment by dcbw** [comment]
@k82cn @luxas PTAL, thanks!

File: pkg/util/iptables/iptables_linux.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// +build linux
```
Comment (vishh) [inline pkg/util/iptables/iptables_linux.go]: Are the build tags necessary even with explicit naming?

Comment (vishh) [inline pkg/util/iptables/iptables_linux.go]: https://golang.org/pkg/go/build/#hdr-Build_Constraints

**Comment by vishh** [comment]
@dcbw a minor nit. Otherwise this PR is ready to be merged

**Comment by k82cn** [comment]
@dcbw , the platform part is OK to me; delegate lock/iptables part to other reviewers.

File: pkg/util/iptables/iptables_linux.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+/*
+Copyright 2014 The Kubernetes Authors.
```
Comment (luxas) [inline pkg/util/iptables/iptables_linux.go]: Now 2017 ;)

**Review by luxas — APPROVED** [review]
LGTM, thanks for fixing!
There are some nits but they are easily fixable. 

And you found the right cross-build command already, perfect!

**Comment by dcbw** [comment]
@luxas fixed the dates, PTAL thanks!

**Review by luxas — APPROVED** [review]
LGTM, no harm done although there are both `_linux.go` and `build linux`

**Pull request merged**